### DISCUSSION TITLE
Reference to Rule object in RHS method.

### DISCUSSION THREAD

**Discussion by senthiljayapalan** [evrete/evrete#62] [Q&A] [answered]
I am using Annotated Java Class for defining the rules.  I need refer the ruleId and salience mentioned in the @Rule annotation within the RHS method implementation.  Is there any way to do it, other than hard coding it.

Below is the exanoke

	@Rule(value = "ACT-1001", salience = 10)
	@Where(value = {"$today.count >100000"})
	public void someRule(@Fact("$today") Population today, @Fact("$r") RuleEvaluationResults results) {
		RuleResult result = new RuleResult.Builder("ACT-1001") //This is ruleId mentioned in annotation
		                            .setPriority(10) //This is salience mentioned in annotation
				            .addMessageDataPoint("Population.count", String.valueOf(today.getCount()))
				            .build();

		results.addRuleProcessingResult(result);

	}

**Comment by andbi** [comment, marked as answer]
Yes, that's possible. Just include additional `RhsContext` param in your method declaration as described in https://evrete.org/docs/ajr/ , Section "Annotation reference". From there, you will be able to get any runtime information, including the rule's runtime. 

Regards,
@andbi

**Reply by senthiljayapalan** [reply]
Thank you for quick response!..
